mne/time_frequency/tests/test_psd.py

```diff
@@ -52,6 +53,124 @@ def test_psd():
 
 
 @requires_version('scipy', '0.12')
+def test_psd_welch():
+    """Tests the welch PSD
+    """
+    raw = io.Raw(raw_fname)
+    exclude = raw.info['bads'] + ['MEG 2443', 'EEG 053']  # bads + 2 more
+    include = []
+
+    tmin, tmax = 0, 10  # use the first 60s of data
+    fmin, fmax = 2, 70  # look at frequencies between 5 and 70Hz
+    n_fft = 128
+
+    # picks MEG gradiometers
+    picks = pick_types(raw.info, meg='grad', eeg=False, eog=True,
+                       stim=False, include=include, exclude=exclude)
+
+    # Test raw
+    kws_welch = dict(tmin=tmin, tmax=tmax, fmin=fmin, fmax=fmax, n_fft=n_fft)
+    psds, freqs = psd_welch(raw, proj=False, picks=picks, **kws_welch)
+    psds_proj, freqs_proj = psd_welch(raw, proj=True, picks=picks, **kws_welch)
+
+    assert_array_almost_equal(psds, psds_proj)
+    assert_true(psds.shape == (len(picks), len(freqs)))
+    assert_true(np.sum(freqs < 0) == 0)
+    assert_true(np.sum(psds < 0) == 0)
+
+    # Test epochs
+    events = read_events(event_fname)
+    tmin, tmax, event_id = -0.5, 0.5, 1
+    epochs = Epochs(raw, events[:10], event_id, tmin, tmax, picks=picks,
+                    baseline=(None, 0),
+                    reject=dict(grad=4000e-13, eog=150e-6), proj=False,
+                    preload=True)
+
+    tmin_full, tmax_full = -1, 1
+    epochs_full = Epochs(raw, events[:10], event_id, tmin_full, tmax_full,
+                         picks=picks, baseline=(None, 0),
+                         reject=dict(grad=4000e-13, eog=150e-6), proj=False,
+                         preload=True)
+
+    picks_psd = pick_types(epochs.info, meg=True, eeg=True, ref_meg=False,
+                           exclude='bads')
+
+    kws_welch = dict(fmin=2, fmax=300, n_fft=512, picks=picks_psd)
+    psds, freqs = psd_welch(epochs[:1], proj=False, **kws_welch)
+    psds_full, freqs_full = psd_welch(epochs_full[:1], proj=False,
+                                      tmin=tmin, tmax=tmax, **kws_welch)
+    psds_proj, freqs_proj = psd_welch(epochs[:1], proj=True, **kws_welch)
+
+    # this one will fail if you add for example 0.1 to tmin
+    assert_array_almost_equal(psds, psds_full, 27)
+    assert_array_almost_equal(psds, psds_proj, 27)
+
+    assert_true(psds.shape == (1, len(picks_psd), len(freqs)))
+    assert_true(np.sum(freqs < 0) == 0)
+    assert_true(np.sum(psds < 0) == 0)
+
+
+@requires_version('scipy', '0.12')
+def test_psd_multitaper():
```

Comment:
the test_psd functions were calling the `compute_XXX_psd` functions, so I decided to add new tests rather than modify the old ones since the code is going to stick around for a little bit until it's deprecated. I decided to make two separate tests for the two psd methods, but could try to consolidate if need be